agent/hcp/telemetry/filter.go

```diff
@@ -0,0 +1,42 @@
+package telemetry
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/go-multierror"
+)
+
+// FilterList holds a map of filters, i.e. regular expressions.
+// These filters are used to identify which Consul metrics can be transmitted to HCP.
+type FilterList struct {
+	filters map[string]*regexp.Regexp
+}
+
+// NewFilterList returns a FilterList which holds valid regex
+// used to filter metrics. It will not fail if invalid REGEX is given, but returns a list of errors.
+func NewFilterList(filters []string) (*FilterList, error) {
```

Comment:
```suggestion
// newFilterList returns a FilterList which holds valid regex
// used to filter metrics. It will not fail if invalid REGEX is given, but returns a list of errors.
func newFilterList(filters []string) (*FilterList, error) {
```